submissions/shramkoweb/js-dom/main.js

```diff
@@ -0,0 +1,104 @@
+const frontendFrameworks = [
+    {
+        id: 0,
+        name: 'React',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg',
+    },
+    {
+        id: 1,
+        name: 'Vue',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/9/95/Vue.js_Logo_2.svg',
+    },
+    {
+        id: 2,
+        name: 'Angular',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/c/cf/Angular_full_color_logo.svg',
+    },
+    {
+        id: 3,
+        name: 'Django',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/7/75/Django_logo.svg',
+    },
+    {
+        id: 4,
+        name: 'Ruby',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/f/f1/Ruby_logo.png',
+    },
+    {
+        id: 5,
+        name: 'Flask',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/3/3c/Flask_logo.svg',
+    },
+    {
+        id: 6,
+        name: 'Express',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/6/64/Expressjs.png',
+    },
+    {
+        id: 7,
+        name: 'JQuery',
+        url: 'https://upload.wikimedia.org/wikipedia/commons/8/81/JQuery_logo_text.svg',
+    },
+];
+
+const getItemByID = (items, id) => items.find(item => item.id === id);
+const clearHTML = node => node.innerHTML = '';
+
+const renderListItem = ({ name, id }, activeID) => {
+    const item = document.createElement('li');
+    const button = document.createElement('button');
+    button.innerText = name;
+    button.type = 'button';
+    button.id = id;
+    button.className = activeID === id ? 'active button' : 'button';
```

Comment:
It may be better to separate render of buttons list and toggling of active class. It is also better to use [classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) instead of className

Reply:
If you do the activation from the render, you will have to work with the DOM directly bypassing the state, which is not acceptable in MVC

Reply:
Render knows only about the state, the state changes in the handlers (controller), rendering is based on the state, no information is stored or read in the HOUSE, otherwise the essence of MVC which I use here is lost